idutils/__init__.py

```diff
@@ -181,6 +181,10 @@
 """Matches new style arXiv ID, with an old-style class specification;
     technically malformed, however appears in real data."""
 
+hal_regexp = re.compile("(hal:|HAL:)?([a-z]{3}[a-z]*-|(sic|mem|ijn)_)\d{8}$")
```

Comment:
Do you have public documentation about the HAL identifier? 

I see identifiers on your site like ``hal-00671912v1`` which would not be matched against the regular expression.
Also, is there a controlled vocabulary of the first characters (e.g. ``hal``, ``halshs``, ...)? This would make the matching more specific.